Integrate standalone simulation

flomonster opened the pull request:
Close #416 
Close #437 
Close #439 
Close #477 
Close #493
Close #499 
Close #528 
Close #543

flomonster committed:
core: add standalone simulation batch test

flomonster committed:
all: integrate standalone simulation

multun committed:
core: rollingstock: ensure effort curves are sorted

flomonster committed:
api: standalone simulation add allowances

flomonster committed:
api / core: handle route occupancy for standalone simulation

nicolaswurtz committed:
front: use new standalone simulation endpoint for train schedule creation #489

nicolaswurtz committed:
front: upgrade to the new block occupancy format #491

nicolaswurtz committed:
front: allowances beginning WIP

flomonster committed:
tests: fix using standalone simulations

eckter committed:
core: pathfinding: remove consecutive duplicate routes

flomonster committed:
actions: api clear ci cache

nicolaswurtz committed:
front: allow duplicate trains with new format

nicolaswurtz committed:
front: repair trainDetail bar

nicolaswurtz committed:
front: repair trainDetail for trains without block occupancy

nicolaswurtz committed:
front: better display of tracks, popup on OP

multun committed:
core: {envelope,api}: implement allowances for standalone simulation

flomonster committed:
core: fix pathfinding multiple waypoints on the same route

Commit:
core: envelopes: fix bug with capacitySpeedLimit not stopping the train

flomonster committed:
core: fix maitain speed with envelopes

multun committed:
core: envelope: ensure the intersection of two segments stays in bound

multun committed:
core: envelope: add a default overlay threshold at 0

multun committed:
core: envelope_sim: mareco: discard empty coasting sections

multun committed:
core: {envelope,api}: run mareco with a 30km/h low speed limit

multun committed:
core: {envelope,api}: test the standalone api with mareco

flomonster committed:
core: standalone fix default time step

multun committed:
core: envelope: delay base curve slicing until add

This allows for creating an overlay and discarding it.
Previously, doing so resulted in data corruption.

flomonster committed:
core: envelopes: fix maintain speed infinite loop

flomonster committed:
api/core: add eco standalone simulation results

multun commented on api/osrd_infra/models/rolling_stock.py:
```diff
@@ -121,8 +110,7 @@ class RollingStock(models.Model):
         help_text=_(
-            "The inertia coefficient. It will be multiplied with the mass "
-            "of the train to get its effective mass"
+            "The inertia coefficient. It will be multiplied with the mass " "of the train to get its effective mass"
```
these formatting changes are quite odd, why do these occur?

flomonster committed:
core: fix train path gradient constrcution

Commit:
core: envelopes: fix capacitySpeedLimit discontinuity bug

flomonster committed:
api: merge migrations

Tguisnet committed:
core: api: route occupancy with conflicted routes

flomonster committed:
load-database: add chartos dump

flomonster committed:
tests: fix envelopes / allowances
  - TrainPhysicsIntegrator fix critical bug
  - MRSP and EnvelopeTrainPath fix constructions
  - Coasting trigger detection
  - ...

flomonster committed:
api: fix simulation report tail positions

flomonster committed:
core: envelopes: fix threshold
